staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -251,6 +252,26 @@ func (s *Validator) validateExpressions(ctx context.Context, fldPath *field.Path
 			continue
 		}
 		if evalResult != types.True {
+			tryMessage := true
```

Comment:
This function is getting out of hand. Can we split some of this message related logic into a utility?

Reply:
Right, I've added `evalMessageExpression`.